src/renderers/1d/symbol_triangle.cpp

```diff
@@ -0,0 +1,67 @@
+#include "renderers/1d/symbol_triangle.h"
+
+using namespace Mildred;
+
+TriangleSymbolRenderer1D::TriangleSymbolRenderer1D(Qt3DCore::QEntity *rootEntity) : SymbolRenderer1D(rootEntity)
+{
+    symbols_ = new LineEntity(rootEntity_);
+}
+
+TriangleSymbolRenderer1D::~TriangleSymbolRenderer1D()
+{
+    if (symbols_)
+    symbols_->setParent(static_cast<Qt3DCore::QNode *>(nullptr));
+};
+
+/*
+ * Rendering
+ */
+
+// Create entities from the supplied metrics and data
+void TriangleSymbolRenderer1D::create(const ColourDefinition &colour, const std::vector<double> &x, const AxisEntity *xAxis,
+                                const std::vector<double> &values,
+                                const AxisEntity *valueAxis)
+{
+    assert(symbols_);
+    symbols_->clear();
+
+    colour_ = colour;
+
+    // Check array sizes
+    if (x.size() != values.size())
+    {
+        printf("Irregular vector sizes provided (%zu (x) vs %zu (y)) so can't create entities.\n", x.size(),
+               values.size());
+        symbols_->finalise();
+        return;
+    }
+
+    // Loop over data and add vertices
+    auto xit = x.cbegin(), vit = values.cbegin();
+    auto i = 0;
+    double w = symbolMetric_/2.0;
```

Comment:
It's a good habit to use `auto` when doing assignments and the type of the result is deducible from the right-hand side:
```suggestion
    auto w = symbolMetric_/2.0;
```